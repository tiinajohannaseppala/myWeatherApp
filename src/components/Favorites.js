import React from 'react';
import PropTypes from 'prop-types';

const Favorites = ({favorites, deleteFavorite, showFavoriteWeather}) => {

  const FavoriteList = favorites.length ? (

    favorites.map(favorite => {
      return (        
          <div className="favorite" key={favorite.id}>
            <button className="favorites" onClick={() => {showFavoriteWeather(favorite.city)}}>{ favorite.city }</button>
            <button className="favoritesD" onClick={() => {deleteFavorite(favorite.id)}}>x</button>
          </div>
    )}
    )
    ) : (
        <p className="favoritesAlert">Hey, you have no favorites</p>
    )

  return (
    <div className="favorite-list">
      {FavoriteList} 
    </div>
  )
}

// FOR TESTING ---------------------------------------

Favorites.propTypes = {
  favorites: PropTypes.array,
};

Favorites.defaultProps = {
  favorites: [],
};
// ----------------------------------------------------

export default Favorites