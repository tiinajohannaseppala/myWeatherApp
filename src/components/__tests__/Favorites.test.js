import React from 'react';
import renderer from 'react-test-renderer';
import Favorites from '../Favorites';

it('renders when no favorites', () => {
  const tree = renderer.create(<Favorites />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when multiple favorites', () => {
    const favorites = [
    {id: 1, city: 'Helsinki'},
    {id: 2, city: 'Kemi'},
    {id: 3, city: 'Kolari'}];
    const tree = renderer.create(<Favorites favorites={favorites} />).toJSON();
    expect(tree).toMatchSnapshot();
  });