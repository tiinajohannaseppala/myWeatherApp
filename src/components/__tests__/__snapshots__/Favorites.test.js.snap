// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when multiple favorites 1`] = `
<div
  className="favorite-list"
>
  <div
    className="favorite"
  >
    <button
      className="favorites"
      onClick={[Function]}
    >
      Helsinki
    </button>
    <button
      className="favoritesD"
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div
    className="favorite"
  >
    <button
      className="favorites"
      onClick={[Function]}
    >
      Kemi
    </button>
    <button
      className="favoritesD"
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div
    className="favorite"
  >
    <button
      className="favorites"
      onClick={[Function]}
    >
      Kolari
    </button>
    <button
      className="favoritesD"
      onClick={[Function]}
    >
      x
    </button>
  </div>
</div>
`;

exports[`renders when no favorites 1`] = `
<div
  className="favorite-list"
>
  <p
    className="favoritesAlert"
  >
    Hey, you have no favorites
  </p>
</div>
`;
